metabase/metabase
# Extract tokenizer error treatment function in ExpressionEditorTextfield

Chipping away impediments to get to https://github.com/metabase/metabase/issues/18647

Changes

**1.** 
This editor's change handler function does many things.
Originally so much of it is just this one long function doing 10 different things.

As react-ace will offer accessors for a few of the things we manually retrieve today, let's encapsulate things a bit so we hopefully can cleanly throw a few of them out soon, and neatly use the now-organized remainders.

————

**2.** 

A few component’s dirs and file structures are organized.

### How to test

1. Ask a question
2. Custom question
3. Sample Dataset (or any db)
4. People (or any table)
5. Custom column icon
6. Fill input with an incomplete expression like `concat(["Title"`
7. Blur from input

You should see validation messages below the input.
